src/inference/src/cpp_interfaces/interface/ie_iplugin_internal.cpp

```diff
@@ -323,7 +323,26 @@ void SetExeNetworkInfo(const std::shared_ptr<IExecutableNetworkInternal>& exeNet
     const auto& inputsInfo = exeNetwork->GetInputsInfo();
     const auto& outputsInfo = exeNetwork->GetOutputsInfo();
     OPENVINO_ASSERT(inputsInfo.size() == function->get_parameters().size());
-    OPENVINO_ASSERT(outputsInfo.size() == function->get_output_size());
+
+    if (outputsInfo.size() != function->get_output_size()) {
```

Comment:
Can we use `function->outputs().size()`?

Reply:
As `function->outputs()` has additional loop with conversion to `<ov::Output<ov::Node>>` ,
I think just for a size check `function->get_output_size()` can be more efficient.